go.mod

```diff
@@ -1,10 +1,12 @@
 module github.com/inspektor-gadget/inspektor-gadget
 
-go 1.21
+go 1.21.0
+
+toolchain go1.22.0
```

Comment:
Does this means go will download `1.22.0` and compile everything using it? 

Also, in past we wanted to set [`GOTOOLCHAIN=local` in CI/Makefile](https://github.com/inspektor-gadget/inspektor-gadget/pull/2206#issuecomment-1807792513) is that be still relevant?